### PR TITLE
update autorest pr

### PR DESCRIPTION
**Pull request by m-nash**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by pakrym — APPROVED** [review]

File: eng/Directory.Build.Common.targets [hunk 1]
```diff
@@ -198,6 +199,12 @@
     <Compile Include="$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs" LinkBase="Shared" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(IsClientLibrary)' == 'true' and '$(IsTestProject)' != 'true'">
```
Comment (pakrym) [inline eng/Directory.Build.Common.targets:202]: @AlexanderSher can you please move this into individual project files after the PR goes in.

File: sdk/storage/Azure.Storage.Blobs.Batch/src/Azure.Storage.Blobs.Batch.csproj [hunk 2]
```diff
@@ -34,6 +35,9 @@
     <Compile Include="$(AzureCoreSharedSources)HttpMessageSanitizer.cs" LinkBase="Shared\Core" />
     <Compile Include="$(AzureCoreSharedSources)DiagnosticScopeFactory.cs" LinkBase="Shared\Core" />
     <Compile Include="$(AzureCoreSharedSources)TaskExtensions.cs" LinkBase="Shared\Core" />
+    <Compile Include="$(AzureCoreSharedSources)OperationInternal.cs" LinkBase="Shared" />
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs.Batch/src/Azure.Storage.Blobs.Batch.csproj:38]: Are these duplicated now that you've added them to Common.props?

**Review by pakrym — APPROVED** [review]

**Review by pakrym — APPROVED** [review]

**Comment by heaths** [comment]
@m-nash after many failed attempts using github.dev (stable or release channels), I'm unable to review this. It hangs my brother on github.com and never renders a PR view in github.dev.

Can you summarize the change here, or link to the autorest / autorest.csharp PR(s) for this change?

**Comment by annelo-msft** [comment]
@heaths, VS Code has a nice interface for viewing large PRs like this: https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github

**Comment by m-nash** [comment]
@heaths The changes relevant to dataplane are, I will try to keep this list updated as there are going to be a few more dataplane projects that need updated

- [update dataplane libraries to use 3 new files](https://github.com/Azure/azure-sdk-for-net/pull/25492/commits/bd46432dab9b70cb8c43647c02df4544d9f4a8ce)
- [update azure core experimental](https://github.com/Azure/azure-sdk-for-net/pull/25492/commits/8159421be0d2a67c48e44e542ffa6f2649ee1e0c)
- [update dataplane keyvault](https://github.com/Azure/azure-sdk-for-net/pull/25492/commits/0d3889b19c50e4f19a7f3212385c40d69032f266)
- [remove 429 restriction on playback transport](https://github.com/Azure/azure-sdk-for-net/pull/25492/commits/02e06e6aa38eafe55ffe4f98082ae563ef5528eb)
- [Only add retry-after once](https://github.com/Azure/azure-sdk-for-net/pull/25492/commits/2a9dfdb8cb3a6a8c3fc709ac2f45a20100f3063e)

**Comment by heaths** [comment]
> @heaths, VS Code has a nice interface for viewing large PRs like this: https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github

@annelo-msft @pakrym and I have both been trying to use it. It's been very inconsistent, and is what I normally use for large PRs to get something almost as good as AzDO's PR workflow. Just not in the cards this afternoon. Worked for Josh's large (about the same number of files impacted) PR this morning just fine.

Tried the latest version of the extension on both the stable and insider channels for VSCode to no avail.

**Comment by heaths** [comment]
I did add a [comment](https://github.com/microsoft/vscode-pull-request-github/issues/2741#issuecomment-982188081) to an open issue about large PRs using the extension to this effect with a video of the repro I'm seeing.

File: eng/Directory.Build.Common.targets (hunk 1 above)
Comment (AlexanderSher) [reply eng/Directory.Build.Common.targets]: Ok

**Review by AlexanderSher — APPROVED** [review]
LGTM

**Review by fengzhou-msft — COMMENTED** [review]
Mgmt plane LGTM.
